[vulkan, spirv, etc] Update to v1.4.328

tophyr opened the pull request:
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

tophyr committed:
Update Vulkan, SPIRV, etc to v1.4.321.0

tophyr committed:
patch ggml until https://github.com/ggml-org/ggml/pull/1317

tophyr committed:
fixup lunarg-vulkantools portfile

dg0yt commented on ports/vulkan-extensionlayer/portfile.cmake:
```diff
@@ -25,4 +25,11 @@ if(VCPKG_TARGET_IS_ANDROID)
+if(EXISTS "${SOURCE_PATH}/LICENSE")
+    list(APPEND LICENSE_FILES "${SOURCE_PATH}/LICENSE")
+endif()
+vcpkg_install_copyright(FILE_LIST ${LICENSE_FILES})
```
I hope that devendored dependencies are already removed at this point.
And every file called LICENSE here. So will be the headlines in the copyright file.
I sometimes copy the files to more verbose names, E.g.
https://github.com/microsoft/vcpkg/blob/0804e3b55b7e435c593707071052363fa876f170/ports/duckdb/portfile.cmake#L148-L167

tophyr replied:
I'm not quite understanding this comment - the `vulkan-extensionlayer` license file structure is thus:
```
src
 |- LICENSES
     |- Apache-2.0.txt
     |- CC-BY-4.0txt
     |- LLVM-exception.txt
     |- MIT.txt

(there is no src/LICENSE)
```

this is certainly an uncommon setup (and i actually reported a bug about it, i think, but was basically shoo'ed off iirc) but i'm not seeing how your suggested snippet is materially different from what's there

dg0yt replied:
Sorry, my bad. Probably too focused on the "/LICENSE" tail :facepalm:

tophyr replied:
hah, that's ok, going through the full diff changes here i only just remembered that _i'm the one who wrote_ this snippet in the first place lol. but yeah, this appears to be how the upstream project wants the licenses tracked.

tophyr committed:
patch-fix directfb support in vulkan-loader

tophyr committed:
skip vulkan-loader[directfb] in CI since machines don't have its necessary libs installed

tophyr committed:
fix vulkan-loader git hash

tophyr committed:
fix lunarg-vulkantools compilation for windows-static-md

tophyr committed:
fix versions again

tophyr committed:
WIP: update vulkan stuff to 1.4.328

tophyr committed:
remove unnecessary diff

tophyr committed:
use parallel-hashmap instead of robin-hood-hashing in vulkan-validationlayers

tophyr committed:
Merge branch 'update-to-1.4.328' into fixup-smaller-upgrade

tophyr commented on scripts/ci.feature.baseline.txt:
```diff
@@ -289,6 +289,7 @@ usbmuxd:x64-windows-static-md=fail
 v8:x64-osx=fail
 v8(windows)=fail # the version of v8 we have in the repo doesn't support VS2022
 vectorclass:arm64-windows=fail
+vulkan-loader[directfb](linux)=feature-fails # vcpkg CI machines don't have libdirectfb installed
```
@dg0yt you seem pretty knowledgeable: any ideas on how we could get the Linux CI machines to include a `libdirectfb` package?

dg0yt replied:
Wikipedia: https://en.wikipedia.org/wiki/DirectFB
So there is directfb2. 
But why not omit the directfb feature for now.

tophyr commented:
@dg0yt marking this as ready-for-review.

tophyr committed:
add spirv-diff to spirv-tools tools list

tophyr committed:
eliminate vulkantools static linkage configuration and instead depend on vcpkg

tophyr committed:
update versions, sigh

tophyr committed:
Merge branch 'master' into fixup-smaller-upgrade

tophyr committed:
[vulkan-validationlayers] disable use of vendored parallel_hashmap

vcpkg guidelines require that ports have exactly one dependency provider. the upstream source here attempts to autodetect a system-provided parallel_hashmap library, and builds its own if one is not present. while convenient for upstream, this results in different sources being available. in this diff we patch the upstream source to remove the vendored copy and streamline the cmake logic for finding it.

tophyr committed:
Merge branch 'master' into fixup-smaller-upgrade

tophyr committed:
dunno how these damn versions keep being wrong

tophyr committed:
Merge branch 'master' into fixup-smaller-upgrade

tophyr committed:
vulkan-sdk-components does not depend on robin-hood-hashing

ajtribick commented on ports/vulkan-validationlayers/disable_vendored_phmap.diff:
Is it necessary to delete all the vendored files (resulting in a fairly huge diff file), or is removing the references from the CMakeLists.txt file sufficient here?

vicroms replied:
Agree with dg0yt here.

Keep the changes to `CMakeList.txt` but instead of removing the `parallel-hashmap` files via a patch, run `file(REMOVE_RECURSE "${SOURCE_PATH}/layers/external/parallel-hashmap")` in the portfile.

vicroms commented on versions/s-/spirv-headers.json:
```diff
@@ -1,5 +1,15 @@
+      "git-tree": "d341a149c9b2ebf990e3a933d967a22954ba4d8d",
+      "version": "1.4.321.0",
+      "port-version": 0
+    },
```
```suggestion
```

vicroms commented on versions/s-/spirv-reflect.json:
```diff
@@ -1,5 +1,15 @@
+      "git-tree": "349969474da413fd150123ecfd5eba82bb88c1cb",
+      "version": "1.4.321.0",
+      "port-version": 0
+    },
```
```suggestion
```

vicroms commented on versions/v-/vk-bootstrap.json:
```diff
@@ -1,5 +1,15 @@
+      "git-tree": "f60095626947e358812495ae79e51c10b734efa6",
+      "version": "1.4.321",
+      "port-version": 0
+    },
```
```suggestion
```

vicroms commented on versions/v-/volk.json:
```diff
@@ -1,5 +1,15 @@
+      "git-tree": "c064c03fb5038b16d360f523c31099ee06780b1e",
+      "version": "1.4.321.0",
+      "port-version": 0
+    },
```
```suggestion
```

vicroms commented on versions/v-/vulkan-validationlayers.json:
```diff
@@ -1,5 +1,15 @@
+      "git-tree": "72a608f69e45b9d2993ea0c5f4b170a24e3fc3f3",
+      "version": "1.4.321.0",
+      "port-version": 0
+    },
```
```suggestion
```

vicroms commented on versions/v-/vulkan-utility-libraries.json:
```diff
@@ -1,5 +1,15 @@
+      "git-tree": "68d1625f6a280ac06ba900890a52140b89b0b60b",
+      "version": "1.4.321.0",
+      "port-version": 0
+    },
```
```suggestion
```

vicroms commented on versions/v-/vulkan-loader.json:
```diff
@@ -1,5 +1,15 @@
+      "git-tree": "9720b4822a220b3ab15c8a9cba5366f948b3b9d1",
+      "version": "1.4.321.0",
+      "port-version": 0
+    },
```
```suggestion
```

vicroms commented on versions/v-/vulkan-headers.json:
```diff
@@ -1,5 +1,15 @@
+      "git-tree": "870e7615b3b8805a00a0465b4dda630bae4a20b7",
+      "version": "1.4.321.0",
+      "port-version": 0
+    },
```
```suggestion
```

tophyr committed:
way better devendoring approach; eliminate spurious versions

tophyr commented on versions/l-/lunarg-vulkantools.json:
```diff
@@ -1,5 +1,15 @@
+      "git-tree": "72771d3132aad0c9b058b01db605feb31b7ad1dc",
+      "version": "1.4.321.0",
+      "port-version": 0
+    },
```
Tangent: I've wondered about this in the past - I assume that the `REF "vulkan-sdk-${VERSION}"` pattern in portfiles is so that vcpkg downloads the correct hash/tag/etc when a version is explicitly specified. However, I've never seen a portfile actually check for versions beyond using it to form the ref. It seems like it would be fairly common to have portfile changes during a version update (say, adding a new patch to apply) that would cause older versions to fail compilation if the new portfile were executed with the old ref.

Is there something I'm not seeing that prevents this? Or is this just such an uncommon use case that it basically gets ignored?

dg0yt replied:
The portfile used is always matching the port manifest used. With `${VERSION}` (derived from the manifest) in the portfile, we simply avoid encoding the literal version in a second place (in addition to the manifest). The download is the most frequent use case, but others exist (locating filepaths, informing build options ...)

ajtribick commented:
The issue I see is that it uses the vendored parallel-hashmap natvis file here: https://github.com/KhronosGroup/Vulkan-ValidationLayers/blob/b08b9a92175a2cca3659703ae08d9c033df7a04d/layers/CMakeLists.txt#L503

The parallel-hashmap port does actually install this file, so maybe that can be used instead: https://github.com/microsoft/vcpkg/blob/c50ed424775a4227d3e8a784d0ce5cccfe19caa4/ports/parallel-hashmap/portfile.cmake#L23

tophyr committed:
use provided parallel-hashmap's phmap.natvis file

tophyr committed:
Merge remote-tracking branch 'upstream/master' into fixup-smaller-upgrade

tophyr committed:
previous updates to ggml make our changes unnecessary (and incompatible)

tophyr committed:
vulkan-loader[directfb](linux) should now pass

tophyr committed:
maybe we do still need the dequant fix i guess

tophyr committed:
gixup ggml port-version

tophyr committed:
keep const in ggml patch

tophyr commented:
don't want to merge it in because it looks like it's still in flux, but waiting on this PR for https://github.com/microsoft/vcpkg/pull/48492

tophyr committed:
Merge remote-tracking branch 'upstream/master' into fixup-smaller-upgrade

vicroms reviewed (changes requested):
Almost there, one merge issue on ports version db history.

tophyr committed:
Merge remote-tracking branch 'upstream/master' into fixup-smaller-upgrade

tophyr committed:
undo overwriting version history

tophyr commented:
🥳
